cdap-ui/app/cdap/components/ConfigurationGroup/index.tsx

```diff
@@ -106,36 +111,117 @@ const ConfigurationGroupView: React.FC<IConfigurationGroupProps> = ({
     inputSchema,
   };
 
+  // Used to keep track of error messages that found a widget
+  // required to identify errors that did not find a widget,
+  // they will be marked as orphan errors.
+  const [usedErrors, setUsedErrors] = React.useState({});
+  const [groups, setGroups] = React.useState([]);
+  const [orphanErrors, setOrphanErrors] = React.useState([]);
+  React.useEffect(
+    () => {
+      setGroups(constructGroups);
+    },
+    [configurationGroups, errors]
+  );
+  React.useEffect(
+    () => {
+      setOrphanErrors(constructOrphanErrors);
+    },
+    [usedErrors]
+  );
+  function constructGroups() {
+    const newUsedErrors = {};
```

Comment:
I am not sure I entirely sure if I understand what `newUsedErrors` mean though.  Are we clearing out errors as user starts to fix them? Is that what this tracks to not show error styling on those properties that the user is fixing?

Reply:
That's just a temporary variable to update 'usedErrors' state.